src/figury/AnimPanel.java

```diff
@@ -47,8 +47,22 @@ public void initialize() {
 	}
 
 	void addFig() {
-		Figura fig = (numer++ % 2 == 0) ? new Kwadrat(buffer, delay, getWidth(), getHeight())
-				: new Elipsa(buffer, delay, getWidth(), getHeight());
+		Figura fig;
+		switch(numer++ % 3)
+		{
+			case 0:
+				fig = new Kwadrat(buffer, delay, getWidth(), getHeight());
+				break;
+			case 1:
+				fig = new Elipsa(buffer, delay, getWidth(), getHeight());
+				break;
+			case 2:
+				fig = new Trojkat(buffer, delay, getWidth(), getHeight());
+				break;
+			default:
+				fig = null;
```

Comment:
Oj, gdyby ta linia była osiągalna to za chwilę mielibyśmy wyjątek przy próbie utworzenia wątku